Implementations for Global Sitecore.Foundation.SitecoreExtensions project into Headstart Api solution.

ppatel-sitecore opened the pull request:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.
2. Completed Code clean-up for all classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments, and removing un-used namespace references). This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Headstart.API' project.
3. Completed: Code clean-up for all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments, and removing un-used namespace references).
4. Completed: Adding the implementation of the logging mechanism (try/catch logging) from the 'Sitecore.Foundation.SitecoreExtensions' project to all the classes for the 'Headstart.API' and 'Headstart. Common' projects.
5. Completed: Code clean-up for all classes for the 'Headstart. Common' project to correctly apply coding formatting and standards (i.e. removing any unneeded code comments and removing un-used namespace references). This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Headstart. Common' project.
6. Completed: Merging the latest changes from Crhistian for the PR: https://github.com/ordercloud-api/headstart/pull/361.
7. Completed: Merging the latest changes from Crhistian for the PR: https://github.com/ordercloud-api/headstart/pull/363.
8. Completed: Merging the latest changes from Oliver for the PR: update catalyst version to 2.0.1 for security - https://github.com/ordercloud-api/headstart/pull/364.
9. Added throw exception handling globally for all try/catches as requested by Crhistian.
10. Added Documentation Guide for all the new functionality as requested by Crhistian.

Commit:
1. Add the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism.
2. Update all controller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. code comments and removing un-used namespace references).

Commit:
1. Added a new IsNonProdEnv appSettings configurations.

Commit:
1. Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism.
2. Updated all controller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. code comments and removing un-used namespace references).
3. Updated all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. code comments and removing un-used namespace references).
4. Added the implementation of the logging mechanism (try/catch logging) from the 'Sitecore.Foundation.SitecoreExtensions' project, to all the classes for the 'Headstart.API' project.

Commit:
1. Code clean-up for all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments and removing un-used namespace references).
2. Code clean-up for all classes for the 'Headstart.Common' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments and removing un-used namespace references).

Commit:
1. Updated the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.
2. WIP: Code clean-up for all icontroller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e.  removing unneeded code comments, adding missing code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.API' project.
3. WIP: Code clean-up for all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments and removing un-used namespace references).
4. WIP: Code clean-up for all classes for the 'Headstart.Common' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.API' project.

Commit:
1. WIP: Code clean-up for all icontroller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e.  removing unneeded code comments, adding missing code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.API' project.

Commit:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.
2. Completed Code clean-up for all icontroller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e.  removing unneeded code comments, adding missing code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.API' project.
3. Completed: Code clean-up for all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments and removing un-used namespace references).
4. Completed: Adding the implementation of the logging mechanism (try/catch logging) from the 'Sitecore.Foundation.SitecoreExtensions' project, to all the classes for the 'Headstart.API' and 'Headstart.Common' projects.
5. Completed: Code clean-up for all classes for the 'Headstart.Common' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.Common' project.

Commit:
1. Completed Code clean-up for all classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments, and removing un-used namespace references). This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Headstart.API' project.

Commit:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.

Commit:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.

Commit:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.

Commit:
Removed un-used files

Commit:
Fixing nuget source issues for clean clone and automation build

1. Fixing nuget source issues for clean clone and automation build.

Commit:
Fixing NuGet source issues for clean clone and automation build.

1. Fixing NuGet source issues for clean clone and automation build.

ppatel-sitecore committed:
Update azure-pipelines.yml for Azure Pipelines

1. Fixing NuGet source issues for clean clone and automation build.

Commit:
Merge remote-tracking branch 'ordercloud-api/azure-pipelines' into development

Commit:
Fixing NuGet source issues for clean clone and automation build.

1. Fixing NuGet source issues for clean clone and automation build.
2. Completed Code clean-up for all classes for the 'Headstart.Jobs' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments, and removing un-used namespace references). This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Headstart.Jobs' project.

ppatel-sitecore committed:
Merge branch 'development' into development

Commit:
Added fixes for restoring dependent  Sitecore NuGet packages

1. Added fixes for restoring dependent  Sitecore NuGet packages

Commit:
Merge remote-tracking branch 'Headstart-ppatel/development' into development

Commit:
Added fixes for restoring dependent Sitecore NuGet packages

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

WIP: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

WIP: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
1. Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
1. Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Fixing the headstart-sitecore-foundation-automation-deploy.ps1

Fixing the headstart-sitecore-foundation-automation-deploy.ps1

Commit:
WIP: Resolving UnitTest failures 

1. WIP: Resolving UnitTest failures due to the "Could not load type 'System.Web.Configuration.WebConfigurationManager' from assembly 'System.Web, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'" error.

Commit:
1. WIP: Resolving UnitTest failures due to the "Could not load type 'System.Web.Configuration.WebConfigurationManager' from assembly 'System.Web, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'" error.

Commit:
WIP: Resolving UnitTest failures changing appSettings to json file.

1. WIP: Resolving UnitTest failures due to the "Could not load type 'System.Web.Configuration.WebConfigurationManager' from assembly 'System.Web, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'" error.

Commit:
WIP: Resolving UnitTest failures

1. WIP: Resolving UnitTest failures due to the "Could not load type 'System.Web.Configuration.WebConfigurationManager' from assembly 'System.Web, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'" error.

Commit:
WIP: Resolving UnitTest failures

WIP: Resolving UnitTest failures

Commit:
Moving appSettings logic and configs to the ApiAppSettings.json file

1. Moving appSettings logic and configs to the ApiAppSettings.json file.

Commit:
WIP: Resolving UnitTest failures

WIP: Resolving UnitTest failures

Commit:
WIP: Resolving UnitTest failures

1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

Commit:
WIP: Resolving UnitTest failures

1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

Commit:
WIP: Resolving UnitTest failures

1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

Commit:
1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

Commit:
1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

Commit:
Merging the latest changes from Crhistian for the PR: https://github.com/ordercloud-api/headstart/pull/361

1. Merging the latest changes from Crhistian for the PR: https://github.com/ordercloud-api/headstart/pull/361

Commit:
1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

Commit:
1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

1. WIP: Resolving UnitTest failures, i.e. adding checks for null values before executing test functions that require params that are not null.

Commit:
Update to the PromotionCommandTests.cs making the class public

Update to the PromotionCommandTests.cs making the class public

Commit:
WIP: Resolving the UnitTest failures

1. WIP: Resolving the UnitTest failures.

ppatel-sitecore committed:
Merge branch 'development' into development

Commit:
WIP: Resolving the UnitTest failures

WIP: Resolving the UnitTest failures

Commit:
WIP: Resolving the UnitTest failures

WIP: Resolving the UnitTest failures

Commit:
WIP: Resolving the UnitTest failures

WIP: Resolving the UnitTest failures

Commit:
WIP: Resolving the UnitTest failures

WIP: Resolving the UnitTest failures

Commit:
WIP: Resolving the UnitTest failures

WIP: Resolving the UnitTest failures

Commit:
WIP: Resolving the UnitTest failures

WIP: Resolving the UnitTest failures

Commit:
WIP: Resolving the UnitTest failures

WIP: Resolving the UnitTest failures

Commit:
Completed: Resolving the UnitTest failures

Completed: Resolving the UnitTest failures

Commit:
Resolving missing dependencies for .NET 4.8 and visualstudio2019buildtools

Resolving missing dependencies for .NET 4.8 and visualstudio2019buildtools

Commit:
Resolving missing dependencies for .NET 4.8 and visualstudio2019buildtools

Resolving missing dependencies for .NET 4.8 and visualstudio2019buildtools

Commit:
Resolving missing dependencies for .NET 4.8 and visualstudio2019buildtools

Resolving missing dependencies for .NET 4.8 and visualstudio2019buildtools

Commit:
Fixing issue related to the Microsoft.WebApplication.targets package so it pulls from NuGet package

1. Fixing issue related to the Microsoft.WebApplication.targets package so it pulls from NuGet package.

Commit:
Completed: Merging the latest changes from Crhistian for the PR: Fix sale pricing bugs found by Ashley #363.

1. Completed: Merging the latest changes from Crhistian for the PR: Fix sale pricing bugs found by Ashley #363.

Commit:
1. Completed: Merging the latest changes from Crhistian for the PR: Fix sale pricing bugs found by Ashley #363.

1. Completed: Merging the latest changes from Crhistian for the PR: Fix sale pricing bugs found by Ashley #363.

ppatel-sitecore committed:
Merge branch 'development' into development

Commit:
Fixing Merged changes for the Headstart.Common.SendgridService.cs file.

1. Fixing Merged changes for the Headstart.Common.SendgridService.cs file.

Commit:
Fixing Merged changes for the Headstart.Tests.ValidUnitPriceTests.cs file failed test cases.

1. Fixing Merged changes for the Headstart.Tests.ValidUnitPriceTests.cs file failed test cases.

Commit:
1. Completed: Merging the latest changes from Oliver for the PR: update catalyst version to 2.0.1 for security - https://github.com/ordercloud-api/headstart/pull/364.

ppatel-sitecore committed:
Merge branch 'development' into development

Commit:
1. Added throw exception handling globally for all try/catches as requested by Crhistian.
2. Added Documentation Guide for all the new functionality as requested by Crhistian.

Commit:
Added Documentation Guide for all the new functionality as requested by Crhistian.

Added Documentation Guide for all the new functionality as requested by Crhistian.

Commit:
Added Documentation Guide for all the new functionality as requested by Crhistian.

Added Documentation Guide for all the new functionality as requested by Crhistian.

Commit:
Added Documentation Guide for all the new functionality as requested by Crhistian.

Added Documentation Guide for all the new functionality as requested by Crhistian.

Commit:
Adding lined space in the MD readme file.

Commit:
Fixing choco install for visualstudio2022-workload-vctools

Commit:
Fixing choco install for visualstudio2022-workload-vctools

Commit:
Fixing choco install for visualstudio2022-workload-vctools

Fixing choco install for visualstudio2022-workload-vctools

Commit:
Merge remote-tracking branch 'Headstart.ppatel/development' into development

Commit:
Fixing choco install for visualstudio2022-workload-vctools

Fixing choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Working on Fixing the choco install for visualstudio2022-workload-vctools

Working on Fixing the choco install for visualstudio2022-workload-vctools

Commit:
Fixing False Unit Test cases during after hours and Weekends

1. Fixing False Unit Test cases during after hours and Weekends.

Commit:
Fixing False Unit Test cases during after hours and Weekends

1. Fixing False Unit Test cases during after hours and Weekends.

crhistianramirez commented on README.md:
```diff
@@ -317,3 +317,59 @@ You can run the project using Docker, sample docker-compose.yml file includes Bu
+
+## Sitecore.Foundation.Extentions Project (Now included in main Solution)
+1. This includes build-in extention method functionality for client's wanting to integrate with a Tranditional Sitecore XM/XP instance implementation 
+    - These extension methods can be access via the following files under the src\Middleware\src\SitecoreExtensions\code folder:
```
Thanks for adding this documentation. Would you mind running this through a spellcheck and grammar tool? I'm seeing quite a few errors. For example:

- build-in should be built-in
- extention should be extension
- client's should be clients
- Tranditional should be Traditional

ppatel-sitecore replied:
Resolved the BAC-3584 issue and fixing Seo meta tagging for products

ppatel-sitecore replied:
Resolved this request.

crhistianramirez commented on src/Middleware/src/SitecoreExtensions/code/Extensions/SitecoreLoggingExtensions.cs:
```diff
@@ -0,0 +1,343 @@
+			if (isError)
+			{
+				Log.Error(message, new object());
+				throw new Exception(message);
```
Can you please rethrow the original exception instead of throwing a new exception? The former retains the original stacktrace, while this one does not. Should be simply `throw;` instead of `throw new Exception(message);`;

Details here: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2200

ppatel-sitecore replied:
Resolved this request.

Commit:
Removing temp Fix for False Unit Test cases during after hours and Weekends

1.  Removing temp Fix for False Unit Test cases during after hours and Weekends.
2.  Applying permanent fix for re-throw exception issue in for Unit test cases.
3.  Applying grammar fixes to the ReadMe.md file for the new documentation.

Commit:
Merge remote-tracking branch 'Headstart.ppatel/development' into development
